core/src/services/swift/core.rs

```diff
@@ -0,0 +1,237 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::Debug;
+
+use bytes::Bytes;
+use http::Request;
+use http::Response;
+use serde_json::json;
+
+use crate::raw::*;
+use crate::*;
+
+pub struct SwiftCore {
+    pub root: String,
+    pub endpoint: String,
+    pub account: String,
+    pub container: String,
+    pub token: String,
+    pub client: HttpClient,
+}
+
+impl Debug for SwiftCore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("SwiftCore")
+            .field("root", &self.root)
+            .field("endpoint", &self.endpoint)
+            .field("account", &self.account)
+            .field("container", &self.container)
+            .field("token", &self.token)
```

Comment:
Please don't leak token.

Reply:
Done, but I discovered there're some service also implement this way, like onedrive: https://github.com/apache/incubator-opendal/blob/268d5b8784ce9f30e9c0b70dc6201107a0d5b8f1/core/src/services/onedrive/backend.rs#L58

Reply:
We should fix that. Would you like submit a new PR for this?